File: test-dht22.py
import Adafruit_DHT
sensor = Adafruit_DHT.DHT22
pin = 4  # หรือ 17 ถ้าเปลี่ยนขา
humidity, temperature = Adafruit_DHT.read_retry(sensor, pin)
print('Temp={0:0.1f}C  Humidity={1:0.1f}%'.format(temperature, humidity))

File: test-dht22.js
const sensor = require('node-dht-sensor');
function readDHT() {
  sensor.read(22, 4, function(err, temperature, humidity) {
    if (!err) {
      console.log(`Temp: ${temperature}°C, Humidity: ${humidity}%`);
    } else {
      console.error('Read error:', err);
      setTimeout(readDHT, 2000);
    }
  });
}
readDHT();